codebase2/codebase-sqlite/U/Codebase/Sqlite/Operations.hs

```diff
@@ -877,10 +565,13 @@ s2cBranch (S.Branch.Full.Branch tms tps patches children) =
           boId <- Q.expectBranchObjectIdByCausalHashId chId
           expectBranch boId
 
-saveRootBranch :: C.Branch.CausalBranch Transaction -> Transaction (Db.BranchObjectId, Db.CausalHashId)
-saveRootBranch c = do
+saveRootBranch ::
+  HashHandle ->
```

Comment:
Kind of a bummer how HashHandle has to be passed through these functions that don't use it, just because the low level functions are generic and might need it in some other contexts.

Reply:
> Ah, so some paths do not need to check if they should flush the temp entity table?

I think all paths do need to check if they should flush the temp entity table